price-feeder/oracle/provider/binance_test.go

```diff
@@ -112,4 +112,19 @@ func TestBinanceProvider_GetTickerPrices(t *testing.T) {
 		require.Error(t, err)
 		require.Nil(t, prices)
 	})
+
+	t.Run("check_redirect", func(t *testing.T) {
```

Comment:
Looks like the `check_redirect`s are the same function written multiple time, maybe define this in a separate file and pass it into `t.Run` instead?

Reply:
The problem is that I need the provider to set the `provider.client = server.Client()`, so this way I would need to create one function for each provider (BinanceProvider, HuobiProvider, KrakenProvider, OsmosisProvider) 

I could put all the check redirect tests in one file and create all the providers there, what do you think?

Reply:
Ahh yeah you're right - imo the best way to do it would be to make a couple of `Provider.SetClient` & `Provider.SetBaseURL` functions and then just iterate over the different providers in a slice, but that seems a bit unnecessary just to put these into one function. 

Seems neater the way it is now 👍